webdriver-spec.html

```diff
@@ -3237,183 +3237,384 @@ <h2 id=element-displayedness>Element Displayedness</h2>
-                        to <code><a href='#status-timeout'>timeout</a></code> and return.</li>
-                        <li>if the <code>window</code> fires an <code>unload</code> event, the WebDriver implementation MUST immediately set the response <a href='#response-status'>status</a> to JavascriptError and return with the error message set to "Javascript execution context no longer exists.".</li>
-                        <li>if the <code>callback</code> function is invoked, then:
-                          <ol>
-                            <li>Let <code>result</code> be the first argument passed to <code>callback</code>.</li>
-                            <li>Set the command's response <a href='#response-status'>status</a> to Success.</li>
-                            <li>Let <code>value</code> be the result of the following algorithm:
-                              <ol>
-                                <li>If <code>result</code> is:
-                                  <ol>
-                                    <li><code>undefined</code> or <code>null</code>, return <code> null</code>.</li>
-                                    <li>a long, boolean, or DOMString, return <code>result</code>.</li>
-                                    <li>a <a href='http://w3c.github.io/dom/#element'>Document <code>element</code></a>, then return the corresponding WebElement for that <a href='http://w3c.github.io/dom/#element'>Document <code>element</code></a>.</li>
-                                    <!-- Should there be a recursion limit? -->
-                                    <li>a <code>Sequence &lt;Node> Nodes</code>, then return the result by recursively applying this algorithm to <code>result</code>. WebDriver implementations SHOULD limit the recursion depth.</li>
-                                    <!-- Should there be a limit? -->
-                                    <li>an object, then return the <code>object</code> created by recursively applying this algorithm to each property in <code>result</code>.</li>
-                                  </ol>
-                                </li>
-                              </ol>
-                            </li>
-                            <li>Set the command's response <a href="#response-value">value</a> to <code>value</code>.</li>
-                          </ol>
-                        </li>
-                        <li>Return the response.</li>
-                      </ol>
-                    </li>
-                  </ol>
-                </section>
-        </section>
-        <section>
-          <h2>Reporting Errors</h2>
-          <p></p>
-        </section>
+            <li><p>Return the result of running the <a>clone an
+                  object</a> algorithm with arguments <var>input
+                  value</var> and <var>seen</var>, and the <a>internal
+                  JSON clone algorithm</a> as the clone algorithm.</li>
+            <!-- TODO: Map and Set objects? -->
```

Comment:
Map to a vanilla object (implies keys converted to strings)
Set to an array; element order is consistent with set iteration order

?